app/src/main/java/com/example/dine_and_donate/HomeFragments/NotificationsAdapter.java

```diff
@@ -0,0 +1,116 @@
+package com.example.dine_and_donate.HomeFragments;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.dine_and_donate.Models.Notifications;
+import com.example.dine_and_donate.R;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.List;
+
+public class NotificationsAdapter extends RecyclerView.Adapter<NotificationsAdapter.ViewHolder> {
+
+    private List<Notifications> mNotifications;
+    private Context mContext;
+    private DatabaseReference mRef;
+    private DatabaseReference mNotificationsRef;
+    private FirebaseDatabase mDatabase;
+
+    //pass in posts array in the constructor
+    public NotificationsAdapter(List<Notifications> notifications) {
+        mNotifications = notifications;
+    }
+
+    //for each row, inflate the layout and cache references into ViewHolder (pass them into ViewHolder class)
+    //only invoked when new row needs to be created
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        mContext = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        mDatabase = FirebaseDatabase.getInstance();
+        mRef = mDatabase.getReference();
+
+        View feedView = inflater.inflate(R.layout.item_notification, parent, false);
+        ViewHolder holder = new ViewHolder((feedView));
+        return holder;
+    }
+
+    //bind the values based on the position of the element, called as a user scrolls down
+    public void onBindViewHolder(final ViewHolder holder, int position) {
+        //get data according to position
+        Notifications notification = mNotifications.get(position);
+        String eventId = notification.getEventId();
+        String yelpId = notification.getYelpId();
+        mNotificationsRef = mRef.child("events").child(yelpId).child(eventId);
+        mNotificationsRef.addListenerForSingleValueEvent(new ValueEventListener() {
```

Comment:
can you explain what this is for? i'm having a little trouble following

Reply:
I'm trying to get the event that the Notification object is notifying the user about. 
The events child is organized like this: under events there are yelp ID's and within each yelp ID (restaurant info) there are **multiple** event IDs. It was organized this way since there can be multiple events at a restaurant. 
For each notification item on the notification page, I have to access information about that specific notified event. 
mNotificationsRef will reference the notified event child.